### DISCUSSION TITLE
Have ABAP2XLSX use the ALV Layout and ALV sort options so that users can set this up in the ALV Layout Editor.

### DISCUSSION THREAD

**Discussion by KevinNeale** [abap2xlsx/abap2xlsx#1284] [Ideas]
Does this option exist already? I think it would be very useful. I have started to write one but want to confirm that it doesn't already exist before I pursue it further.
Any help on this would be appreciated.

**Comment by AlexandreHT** [comment]
Hi, 
If you use bind_alv, please try to use zcl_excel_converter->convert and set converter options such as :
```ABAP
       data(converter) = new zcl_excel_converter( ).
        converter->set_option( value zexcel_s_converter_option( filter = abap_true
                                                                   subtot = abap_true
                                                                   hidenc = abap_undefined "hidden column are not exported (instead of hiddenà
                                                                   ) ).

        converter->convert(
          exporting
            io_alv         = salv
            it_table       = table
*            i_row_int      = i_top
*            i_column_int   = i_left
*            i_table        = i_table
*            i_style_table  = table_style
            io_worksheet   = worksheet
          changing
            co_excel       = result ).
```

**Comment by sandraros** [comment]
It should work. Did you use the latest abap2xlsx version? Concerning `zcl_excel_converter`, the demo programs are `zdemo_excel32` and `zdemo_excel33`.

**Comment by KevinNeale** [comment]
Hi Sandra,

Thanks for your prompt reply. I looked into ZDEMO_EXCEL32 but it seems to require the ALV output to be created first and then the EXCEL as a conversion of that output. I wasn't really able to verify that because copying and pasting the code doesn't give me the ALV_STATUS required. 
However, what I need is to be able to create the EXCEL output in background and email it without using ALV. Of course ALV would need to be an option when calling the program in foreground for setting up the ALV_LAYOUT.
Does that makes sense?

**Reply by JingleHells** [reply]
hi @KevinNeale  

Sorry to comment on an old post, but what you're describing is exactly what I'm after.

Did you come up with a solution?
The converter seems to work if you display the alv, but not if you just apply the layout.

Thank you

**Reply by KevinNeale** [reply]
Hi JingleHells,
I finished up writing it myself. I would put it on github but I find that too confusing. Happy to post the code here but won't have time to support it promptly. Let me know if you are interested.

**Reply by JingleHells** [reply]
> Hi JingleHells, I finished up writing it myself. I would put it on github but I find that too confusing. Happy to post the code here but won't have time to support it promptly. Let me know if you are interested.

Hi Kevin , yes, anything would help 
Thanks a lot!

**Reply by JingleHells** [reply]
@KevinNeale

**Reply by KevinNeale** [reply]
Hi JingleHells,
Please see the attached text file. I think you should be able to paste it into a new class called ZCL_OUTPUT_ALV2EXCEL.
[ZCL_OUTPUT_ALV2EXCEL.txt](https://github.com/user-attachments/files/21763657/ZCL_OUTPUT_ALV2EXCEL.txt)
Let me know how you go.

**Reply by JingleHells** [reply]
Thanks a lot, mate!

**Comment by darnoc312** [comment]
Don't use method set_option if you do not want to change the options permanently for any further call of method convert.
You should better use the parameter IS_OPTION of method convert instead. Please refer to #1238.
